### PR TITLE
Don't use emails when generating targeted ids

### PR DESCRIPTION
**Pull request by matthew-puku**
Aims to make RapidConnect easier to upgrade and/or migrate by freezing a fast-moving piece of state. See first commit message for more details.

### CODE REVIEW

**Review by ArthurZheng — APPROVED** [review]
👍 awesome, nice refactoring there and pure function!

**Review by waldofouche — APPROVED** [review]
Awesome work @matthew-puku!

**Comment by vladimir-mencl-eresearch** [comment]
I like this change - will definitely make things easier, without having to maintain state of all (newly) issued targeted IDs.

We run RapidConnect in a replicated way, so this is particularly appealing to us.

**Comment by matthew-puku** [comment]
@rianniello could you sanity check this when you have a second?

File: rapidconnect/spec/models/attributes_claim_spec.rb [hunk 1]
```diff
@@ -56,11 +56,9 @@
   end
 
   it 'sets the edupersontargetedid attribute' do
```
Comment (rianniello) [inline rapidconnect/spec/models/attributes_claim_spec.rb:58]: Does it make sense to add a spec to cover the case of retaining an existing targeted id? (I.e. Manually stick it in Redis)

**Comment by rianniello** [comment]
Thought about this change in detail. 

I believe this shouldn't cause any problems, and am interested to know the initial reasoning for combining email in here. Perhaps the save/lookup logic was just "good enough" to get it working. 

Your change removes this state going forward which is a brilliant optimisation! LGTM except for spec nit.

File: rapidconnect/spec/models/attributes_claim_spec.rb (hunk 1 above)
Comment (matthew-puku) [reply rapidconnect/spec/models/attributes_claim_spec.rb:58]: yep!

**Pull request merged**